### PR TITLE
chore(cosmwasm): update deployment scripts

### PR DESCRIPTION
**Pull request by jcs47**

### CODE REVIEW

File: cosmwasm/deploy-contract.js [hunk 1]
```diff
@@ -187,6 +193,8 @@ const makeMultisigProverInstantiateMsg = (contractConfig, contracts, { id: chain
         },
     } = contractConfig;
 
+    const separator = domainSeparator || keccak256(Buffer.from(`${chainId}${routerAddress}${env}`));
```
[the remark this answers is not in the document]
Comment (jcs47) [reply cosmwasm/deploy-contract.js:196]: for EVM gateway deployment we can already fetch the domain separator from the contract, which I believe is a safer way to go about

[the remark this answers is not in the document]
Comment (jcs47) [reply cosmwasm/deploy-contract.js:196]: that is what I wanted to use, but judging from the config files available in this repo, they are all set to "Axelarnet"

File: evm/deploy-amplifier-gateway.js [hunk 2]
```diff
@@ -167,11 +168,22 @@ async function deploy(config, chain, options) {
         return;
     }
 
+    const {
+        axelar: { contracts, network, axelarId },
```
Comment (milapsheth) [inline evm/deploy-amplifier-gateway.js:172]:
```suggestion
        axelar: { contracts, chainId, axelarId },
```
to be consistent with other evm chain configs. Also need to add this field to all configs

File: cosmwasm/deploy-contract.js [hunk 3]
```diff
@@ -162,8 +167,16 @@ const makeGatewayInstantiateMsg = ({ Router: { address: routerAddress }, VotingV
...
 };
 
-const makeMultisigProverInstantiateMsg = (contractConfig, contracts, { id: chainId }) => {
+const makeMultisigProverInstantiateMsg = (config, chainName) => {
+    const {
+        axelar: { contracts, network, axelarId },
+        chains: { [chainName]: chainConfig },
+    } = config;
+
+    const {id: chainId } = chainConfig;
```
Comment (milapsheth) [inline cosmwasm/deploy-contract.js:176]:
```suggestion
    const { axelarId: chainId } = chainConfig;
```

File: evm/deploy-amplifier-gateway.js (hunk 2 above)
Comment (jcs47) [reply evm/deploy-amplifier-gateway.js:172]: assuming `chainId` on the config, but deconstructing it still to `network` due to `chainId` being already in use within the function

File: evm/deploy-amplifier-gateway.js [hunk 4]
```diff
@@ -167,11 +168,22 @@ async function deploy(config, chain, options) {
...
+    } = contracts;
     contractConfig.deployer = wallet.address;
     const domainSeparator = await getDomainSeparator(config, chain, options);
+    const expectedDomainSeparator = calculateDomainSeparator(axelarId, routerAddress, network);
     const minimumRotationDelay = Number(options.minimumRotationDelay);
     const salt = options.salt || '';
 
+    if (domainSeparator !== expectedDomainSeparator) {
+        throw new Error(`unexpected domain separator (want ${expectedDomainSeparator}, got ${domainSeparator})`);
+    }
```
Comment (milapsheth) [inline evm/deploy-amplifier-gateway.js:185]: check this in getDomainSeparator perhaps. The check should be made if domainSeparator isn't being overridden, otherwise the override isn't useful if a deterministic value is used

File: .github/workflows/test.yaml [hunk 5]
```diff
@@ -25,6 +25,15 @@ jobs:
       - name: Prepare local.json
         run: |
           echo '{
+            "axelar": {
+              "chainId": "axelar-local-1",
```
Comment (milapsheth) [inline .github/workflows/test.yaml:29]: why is this needed if you're overriding the domain separator via CLI and not computing it?

File: cosmwasm/deploy-contract.js [hunk 6]
```diff
@@ -162,8 +167,16 @@ const makeGatewayInstantiateMsg = ({ Router: { address: routerAddress }, VotingV
     return { router_address: routerAddress, verifier_address: verifierAddress };
 };
 
-const makeMultisigProverInstantiateMsg = (contractConfig, contracts, { id: chainId }) => {
+const makeMultisigProverInstantiateMsg = (config, chainName) => {
+    const {
+        axelar: { contracts, chainId: network, axelarId },
```
Comment (milapsheth) [inline cosmwasm/deploy-contract.js:172]:
```suggestion
        axelar: { contracts, chainId: axelarChainId, axelarId },
```
maybe or `axelarNetworkId`?

File: evm/deploy-amplifier-gateway.js (hunk 2 above)
Comment (milapsheth) [inline evm/deploy-amplifier-gateway.js:172]:
```suggestion
        axelar: { contracts, chainId: axelarChainId, axelarId },
```

**Review by milapsheth — APPROVED** [review]
